<a href="https://colab.research.google.com/github/Elispreng/Project-2/blob/main/Spreng_Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First Choice: Portuguese Wine

## Load Data

In [3]:
# import packages for pandas
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [4]:
# mount drive and load data
from google. colab import drive


In [5]:
filename = '/content/drive/MyDrive/Project 2/wines_SPA.csv'
wine_df = pd.read_csv(filename)

In [6]:
#inspect the data
wine_df.head()


winery           wine  year  rating  num_reviews country  \
0  Teso La Monja          Tinto  2013     4.9           58  Espana   
1         Artadi  Vina El Pison  2018     4.9           31  Espana   
2   Vega Sicilia          Unico  2009     4.8         1793  Espana   
3   Vega Sicilia          Unico  1999     4.8         1705  Espana   
4   Vega Sicilia          Unico  1996     4.8         1309  Espana   

             region   price                  type  body  acidity  
0              Toro  995.00              Toro Red   5.0      3.0  
1    Vino de Espana  313.50           Tempranillo   4.0      2.0  
2  Ribera del Duero  324.95  Ribera Del Duero Red   5.0      3.0  
3  Ribera del Duero  692.96  Ribera Del Duero Red   5.0      3.0  
4  Ribera del Duero  778.06  Ribera Del Duero Red   5.0      3.0

In [7]:
wine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7500 entries, 0 to 7499
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   winery       7500 non-null   object 
 1   wine         7500 non-null   object 
 2   year         7498 non-null   object 
 3   rating       7500 non-null   float64
 4   num_reviews  7500 non-null   int64  
 5   country      7500 non-null   object 
 6   region       7500 non-null   object 
 7   price        7500 non-null   float64
 8   type         6955 non-null   object 
 9   body         6331 non-null   float64
 10  acidity      6331 non-null   float64
dtypes: float64(4), int64(1), object(6)
memory usage: 644.7+ KB


In [8]:
wine_df.shape

(7500, 11)

## Data Source

- This data is from [kaggle](https://www.kaggle.com/datasets/fedesoriano/spanish-wine-quality-dataset?resource=download)

## Data Description

- This data contains a range of information about Spanish dessert wines. The dataset includes information about price, rating, year, region, and descriptive qualities (acidity and body). There are 7,500 entries. 

## Target

- For this analysis, the target is rating on a scale of 1 to 5. 

## Rows

- Each row represents a wine. 

## Classification or Regression

- This is a regression problem, because the prediction involves an integer on a scale of 1 to 5. 

## Features

- There are 11 features (5 numeric and 6 categorical). 


> - Name of winery
- Name of wine
- Year the grapes were harvested
- Average rating
- Number of reviews
- Country of Origin
- Region the wine comes from
- Price in Euros
- Wine variety
- Body (1-5 scale)
- Acidity (1-5 scale)






## Rows: How many

- There are 7,500 rows. 

## Challenges?

- Potential challenges involve the unbalanced data that could  indicate outliers and correlation between various features. Other challenges could include missing data or generalizing for the target. 

# Back-up Choice: Adult Income

#Load and Inspect Data

In [9]:
#load data
filename2 = '/content/drive/MyDrive/Project 2/adult[1].csv'
adult_df = pd.read_csv(filename2)

In [10]:
#look at first 5 rows
adult_df.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [11]:
#look at info
adult_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48842 entries, 0 to 48841
Data columns (total 15 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   age              48842 non-null  int64 
 1   workclass        48842 non-null  object
 2   fnlwgt           48842 non-null  int64 
 3   education        48842 non-null  object
 4   educational-num  48842 non-null  int64 
 5   marital-status   48842 non-null  object
 6   occupation       48842 non-null  object
 7   relationship     48842 non-null  object
 8   race             48842 non-null  object
 9   gender           48842 non-null  object
 10  capital-gain     48842 non-null  int64 
 11  capital-loss     48842 non-null  int64 
 12  hours-per-week   48842 non-null  int64 
 13  native-country   48842 non-null  object
 14  income           48842 non-null  object
dtypes: int64(6), object(9)
memory usage: 5.6+ MB


In [12]:
adult_df.shape


(48842, 15)

## Data Source

- This data is from [kaggle](https://www.kaggle.com/datasets/wenruliu/adult-income-dataset)

## Data Description

- This dataset contains information of income below 50K and above 50K. Other demographic information includes race, gender, marital status, original nationality, and hours per week. 

# Target

- The target is "income."

## Rows

- Each row includes the  demographic information of a person. 

## Classification or Regression

- This dataset is classification analysis because it is  a binary class of below 50K or above 50K. 

## Features

- There are 16 features (5 numeric and 11 nominal categorical).

## Rows: How Many?

- There are 48,842 rows

## Challenges?

- Challenges could include the size of the data set which could affect the analysis as well as consideration of the features. It may be necessary to reduce the number of features. Finally challenges may occur when looking at the hyperparameters and overfitting of the models. 